src/main/java/imsam/mgen/MGen.java

```diff
@@ -129,6 +145,16 @@ public void init() throws IllegalArgumentException, JSONException, IOException {
         logger.debug("Number of States: " + numberOfStates);
         logger.debug("Target State: " + targetState);
         logger.debug("Output File: " + outputFilename);
+        initSubclassParamDefaults();
+    }
+
+
+    protected boolean parseSubclassConfigParam(JSONObject json, String key) {
+        return false;
```

Comment:
I assume this function is going to be fully implemented?

Reply:
This method can optionally be overrided by subclasses. Comments added for clarrification.